src/d3a/models/strategy/predefined_load.py

```diff
@@ -77,20 +77,20 @@ def __init__(self, daily_load_profile,
                          balancing_energy_ratio=balancing_energy_ratio,
                          use_market_maker_rate=use_market_maker_rate)
         self.daily_load_profile = daily_load_profile
-        self.load_profile = {}
 
     def event_activate_energy(self):
         """
         Runs on activate event. Reads the power profile data and calculates the required energy
         for each slot.
         :return: None
         """
-        self.load_profile = read_arbitrary_profile(
+        load_profile = read_arbitrary_profile(
             InputProfileTypes.POWER,
             self.daily_load_profile)
-        self._update_energy_requirement()
+        self._update_energy_requirement(load_profile)
+        del self.daily_load_profile
```

Comment:
I guess that this makes a different when you have a large number of predefined loads, right?

Reply:
Indeed. I am not sure if it is worth it actually, because this memory allocation is visible from the start of the simulation, despite that does not hurt to get rid of the profile since it is not needed.